pkg/cli/run.go

```diff
@@ -101,8 +109,22 @@ func NewRun(c CommandContext) *cobra.Command {
 	if err := cmd.RegisterFlagCompletionFunc("region", newCompletion(c.ClientFactory, regionsCompletion).complete); err != nil {
 		cmd.Printf("Error registering completion function for --region flag: %v\n", err)
 	}
-	cmd.PersistentFlags().Lookup("dangerous").Hidden = true
+	toggleHiddenFlags(cmd, []string{"dangerous", "memory", "target-namespace", "secret", "volume", "region", "publish-all",
+		"publish", "link", "label", "interval", "image", "env", "compute-class", "annotation", "dev", "bidirectional-sync", "advanced"}, true)
+
 	cmd.Flags().SetInterspersed(false)
+	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+		fmt.Println(cmd.Short + "\n")
+		//toggle advanced flags on before printing flags out in cmd.UsageString
+		if slices.Contains(args, "--advanced") {
+			toggleHiddenFlags(cmd, []string{"dangerous", "memory", "target-namespace", "secret", "volume", "region", "publish-all",
+				"publish", "link", "label", "interval", "image", "env", "compute-class", "annotation", "dev", "bidirectional-sync", "advanced"}, false)
+		}
+		fmt.Println(cmd.UsageString())
+		if slices.Contains(args, "--advanced") {
+			fmt.Println(cmd.Long)
+		}
```

Comment:
Little bit of a hack, but:
```suggestion
		if slices.Contains(args, "--advanced") {
			toggleHiddenFlags(cmd, []string{"dangerous", "memory", "target-namespace", "secret", "volume", "region", "publish-all",
				"publish", "link", "label", "interval", "image", "env", "compute-class", "annotation", "dev", "bidirectional-sync", "advanced"}, false)
			defer fmt.Println(cmd.Long)
		}
		fmt.Println(cmd.UsageString())
```

Reply:
yeah I was trying to find a way to avoid the double check..this works